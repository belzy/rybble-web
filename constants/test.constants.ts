
type TTest = {
  SNAPSHOT_STRINGS: Array<string>
};

/**
 * Message strings to be used with testing functions like 'describe' and 'it'.
 * 
 * @example 
 * it(SNAPSHOT_STRINGS[0], () => {
 * 
 * // Your Test...
 * 
 * })
 */
export const TEST_STRINGS: TTest = {
  SNAPSHOT_STRINGS: [
    'should match snapshot'
  ]

};