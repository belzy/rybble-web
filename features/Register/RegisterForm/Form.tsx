import React from 'react';
import {
  Text,
  FormControl,
  FormLabel,
  FormErrorMessage,
  FormHelperText,
  Input,
  InputGroup,
  InputLeftElement,
  VStack,
  Button,
  ButtonGroup,
  IconButton,
} from '@chakra-ui/react'
import { EmailIcon, LockIcon } from '@chakra-ui/icons';
import { FaUserAlt, FaTwitter, FaFacebookF, } from 'react-icons/fa';
import { FcGoogle } from 'react-icons/fc';
import { Link } from '../../../components'

/**
 * 
 * @returns { React.ReactElement } React Element
 */
const NameInput = (): React.ReactElement => {

  return (
  <FormControl>

    <InputGroup>

      <InputLeftElement 
        pointerEvents='none'
        children={ <FaUserAlt color='#cbd5e0' fontSize='20px' /> }
      />

      <Input 
        variant='flushed'
        id='register-name'
        type='text'
        name='register-name'
        placeholder='Name or nickname'
        focusBorderColor='brand.600'
      />
    </InputGroup>

  </FormControl>);
};

/**
 * 
 * @returns { React.ReactElement } React Element
 */
const EmailInput = (): React.ReactElement => {

  return (
    <FormControl>

    <InputGroup>

      <InputLeftElement 
        pointerEvents='none'
        children={ <EmailIcon color='gray.300' fontSize='20px' /> }
      />

      <Input 
        variant='flushed'
        id='register-email' 
        type='email' 
        name='register-email'
        placeholder='Email' 
        focusBorderColor='brand.600'
      />
    </InputGroup>

    </FormControl>);
};

const PasswordInput = (): React.ReactElement => {

  return (
    <FormControl>

    <InputGroup>

      <InputLeftElement 
        pointerEvents='none'
        children={ <LockIcon color='gray.300' fontSize='20px' /> }
      />

      <Input 
        variant='flushed'
        id='register-password' 
        type='password' 
        name='register-password'
        placeholder='Password' 
        focusBorderColor='brand.600'
      />
    </InputGroup>

    </FormControl>);
};

const SocialButtons = (): React.ReactElement => {

  return (
    <ButtonGroup spacing={ 4 }>

      <IconButton 
        size='lg'
        aria-label='Sign up with Facebook'
        icon={ <FaFacebookF color='#3B5998' /> }
      />

      <IconButton 
        size='lg'
        aria-label='Sign up with Google' 
        icon={ <FcGoogle /> }
      />

      <IconButton 
        size='lg'
        aria-label='Sign up with Twitter' 
        icon={ <FaTwitter color='#55ACEE' /> }
      />

    </ButtonGroup>);
};

const Form = (): React.ReactElement => {

  return (
    <form style={{ marginTop: '64px', width: '100%' }}>

      <VStack spacing={ 8 }>
        <NameInput />
        <EmailInput />
        <PasswordInput />

        <Button 
          bgColor='brand.600' 
          color='white' 
          _hover={{ backgroundColor: 'brand.500'}}
          fontSize='sm' 
          isFullWidth={ true }
          size='lg'
        >SIGN UP</Button>

        <Text fontWeight='500' color='GrayText'>OR</Text>

       <SocialButtons /> 

      </VStack>


    </form>
  );
};

export default Form;