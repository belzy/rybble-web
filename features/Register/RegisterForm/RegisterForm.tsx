import React from 'react';
import {
  Heading,
  VStack,
  Text,
  Button,
} from '@chakra-ui/react';
import {
  AuthForm,
  NameInput,
  EmailInput,
  PasswordInput,
  SocialButtonGroup,
} from '../../../components';

const Form = (): React.ReactElement => {

  return (
    <AuthForm>
      <NameInput />
      <EmailInput />
      <PasswordInput />

      <Button 
        bgColor='brand.600' 
        color='white' 
        _hover={{ backgroundColor: 'brand.500' }}
        fontSize='sm' 
        isFullWidth={ true }
        size='lg'
      >SIGN UP</Button>

      <Text fontWeight='500' color='GrayText'>OR</Text>

      <SocialButtonGroup
        buttonIcons={ ['facebook', 'google', 'twitter'] }
      />

    </AuthForm>);
};

const RegisterForm = (): React.ReactElement => {

  return (
    <VStack 
      align='center' 
      justify='center' 
      flex='1'
      width='90%'
      maxWidth='384px'
      m='auto'
    >
      <Text 
        fontSize='3xl' 
        as='h2'
        mb='8px'
      >Create Account</Text>

      <Text textAlign='center'>
        Enter your personal details and start planning<br /> your day
      </Text>

      <Form />
    </VStack>
  );
};

export default RegisterForm;