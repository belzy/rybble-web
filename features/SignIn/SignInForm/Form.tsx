import React from 'react';
import {
  Text,
  FormControl,
  FormLabel,
  FormErrorMessage,
  FormHelperText,
  Input,
  InputGroup,
  InputLeftElement,
  VStack,
  Button,
  ButtonGroup,
  IconButton,
} from '@chakra-ui/react'
import { EmailIcon, LockIcon } from '@chakra-ui/icons';
import { FaUserAlt, FaTwitter, FaFacebookF, } from 'react-icons/fa';
import { FcGoogle } from 'react-icons/fc';
import { Link } from '../../../components'

/**
 * 
 * @returns { React.ReactElement } React Element
 */
const EmailInput = (): React.ReactElement => {

  return (
    <FormControl>

    <InputGroup>

      <InputLeftElement 
        pointerEvents='none'
        children={ <EmailIcon color='gray.300' fontSize='20px' /> }
      />

      <Input 
        variant='flushed'
        id='email' 
        type='email' 
        placeholder='Email' 
        focusBorderColor='brand.600'
      />
    </InputGroup>

    </FormControl>);
};

const PasswordInput = (): React.ReactElement => {

  return (
    <FormControl>

    <InputGroup>

      <InputLeftElement 
        pointerEvents='none'
        children={ <LockIcon color='gray.300' fontSize='20px' /> }
      />

      <Input 
        variant='flushed'
        id='password' 
        type='password' 
        placeholder='Password' 
        focusBorderColor='brand.600'
      />
    </InputGroup>

    </FormControl>);
};

const ForgotPasswordLink = (): React.ReactElement => {

  return (
    <Link 
      href='/forgot-password'
      color='brand.600'
      _hover={{ color: 'brand.500' }}
      fontSize='sm'
      fontWeight='500'
    >FORGOT PASSWORD</Link>
  );
};

const SocialButtons = (): React.ReactElement => {

  return (
    <ButtonGroup spacing={ 4 }>

      <IconButton 
        size='lg'
        aria-label='Sign up with Facebook'
        icon={ <FaFacebookF color='#3B5998' /> }
      />

      <IconButton 
        size='lg'
        aria-label='Sign up with Google' 
        icon={ <FcGoogle /> }
      />

      <IconButton 
        size='lg'
        aria-label='Sign up with Twitter' 
        icon={ <FaTwitter color='#55ACEE' /> }
      />

    </ButtonGroup>);
};

const Form = (): React.ReactElement => {

  return (
    <form style={{ marginTop: '64px', width: '100%' }}>

      <VStack spacing={ 8 }>
        <EmailInput />
        <PasswordInput />
        <ForgotPasswordLink />

        <Button 
          bgColor='brand.600' 
          color='white' 
          _hover={{ backgroundColor: 'brand.500'}}
          fontSize='sm' 
          isFullWidth={ true }
          size='lg'
        >SIGN UP</Button>

        <Text fontWeight='500' color='GrayText'>OR</Text>

       <SocialButtons /> 

      </VStack>


    </form>
  );
};

export default Form;