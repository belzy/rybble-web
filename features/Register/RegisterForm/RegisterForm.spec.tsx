import RegisterForm from './RegisterForm';
import Form from './Form';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TEST_STRINGS } from '../../../constants/test.constants';

const { SNAPSHOT_STRINGS } = TEST_STRINGS;

describe('features/Register/RegisterForm', () => {

  it(SNAPSHOT_STRINGS[0], () => {

    const { container } = render(<RegisterForm />);

    expect(container).toMatchSnapshot();
  });

  it(SNAPSHOT_STRINGS[0], () => {

    const { container } = render(<Form />);

    expect(container).toMatchSnapshot();

  });

  /**
   * TODO: Write tests
   * TODO: The sign up button should be disabled initially.
   * TODO: The sign up button should be enabled when inputs are valid and complete.
   * TODO: Should handle error when user with that email already exists.
   * TODO: Should handle errors when form input values are not valid (name, email, password).
   * 
   */
});