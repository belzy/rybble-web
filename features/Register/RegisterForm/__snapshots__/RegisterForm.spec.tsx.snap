// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/Register/RegisterForm should match snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-1ybhgcp"
  >
    <h2
      class="chakra-text css-1cu7ki0"
    >
      Create Account
    </h2>
    <p
      class="chakra-text css-xi606m"
    >
      To keep connected with us please sign
      <br />
       in with your personal info
    </p>
    <form
      style="margin-top: 64px; width: 100%;"
    >
      <div
        class="chakra-stack css-14witgw"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <div
            class="chakra-input__group css-4302v8"
          >
            <div
              class="chakra-input__left-element css-1u2k0bl"
            >
              <svg
                color="#cbd5e0"
                fill="currentColor"
                font-size="20px"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(203, 213, 224);"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
                />
              </svg>
            </div>
            <input
              class="chakra-input css-0"
              id="name"
              placeholder="Name or nickname"
              type="name"
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <div
            class="chakra-input__group css-4302v8"
          >
            <div
              class="chakra-input__left-element css-1u2k0bl"
            >
              <svg
                class="chakra-icon css-hywlp1"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
                  />
                  <path
                    d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
                  />
                </g>
              </svg>
            </div>
            <input
              class="chakra-input css-0"
              id="email"
              placeholder="Email"
              type="email"
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <div
            class="chakra-input__group css-4302v8"
          >
            <div
              class="chakra-input__left-element css-1u2k0bl"
            >
              <svg
                class="chakra-icon css-hywlp1"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              class="chakra-input css-0"
              id="password"
              placeholder="Password"
              type="password"
            />
          </div>
        </div>
        <button
          class="chakra-button css-18m91un"
          type="button"
        >
          SIGN UP
        </button>
        <p
          class="chakra-text css-1y481mq"
        >
          OR
        </p>
        <div
          class="chakra-button__group css-1rbkfj6"
          role="group"
        >
          <button
            aria-label="Sign up with Facebook"
            class="chakra-button css-14axtgj"
            type="button"
          >
            <svg
              aria-hidden="true"
              color="#3B5998"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(59, 89, 152);"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              />
            </svg>
          </button>
          <button
            aria-label="Sign up with Google"
            class="chakra-button css-14axtgj"
            type="button"
          >
            <svg
              aria-hidden="true"
              enable-background="new 0 0 48 48"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 48 48"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                fill="#FFC107"
              />
              <path
                d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                fill="#FF3D00"
              />
              <path
                d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                fill="#4CAF50"
              />
              <path
                d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                fill="#1976D2"
              />
            </svg>
          </button>
          <button
            aria-label="Sign up with Twitter"
            class="chakra-button css-14axtgj"
            type="button"
          >
            <svg
              aria-hidden="true"
              color="#55ACEE"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(85, 172, 238);"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`features/Register/RegisterForm should match snapshot 2`] = `
<div>
  <form
    style="margin-top: 64px; width: 100%;"
  >
    <div
      class="chakra-stack css-14witgw"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-4302v8"
        >
          <div
            class="chakra-input__left-element css-1u2k0bl"
          >
            <svg
              color="#cbd5e0"
              fill="currentColor"
              font-size="20px"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(203, 213, 224);"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
              />
            </svg>
          </div>
          <input
            class="chakra-input css-0"
            id="name"
            placeholder="Name or nickname"
            type="name"
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-4302v8"
        >
          <div
            class="chakra-input__left-element css-1u2k0bl"
          >
            <svg
              class="chakra-icon css-hywlp1"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
                />
                <path
                  d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
                />
              </g>
            </svg>
          </div>
          <input
            class="chakra-input css-0"
            id="email"
            placeholder="Email"
            type="email"
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          class="chakra-input__group css-4302v8"
        >
          <div
            class="chakra-input__left-element css-1u2k0bl"
          >
            <svg
              class="chakra-icon css-hywlp1"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <input
            class="chakra-input css-0"
            id="password"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <button
        class="chakra-button css-18m91un"
        type="button"
      >
        SIGN UP
      </button>
      <p
        class="chakra-text css-1y481mq"
      >
        OR
      </p>
      <div
        class="chakra-button__group css-1rbkfj6"
        role="group"
      >
        <button
          aria-label="Sign up with Facebook"
          class="chakra-button css-14axtgj"
          type="button"
        >
          <svg
            aria-hidden="true"
            color="#3B5998"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(59, 89, 152);"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            />
          </svg>
        </button>
        <button
          aria-label="Sign up with Google"
          class="chakra-button css-14axtgj"
          type="button"
        >
          <svg
            aria-hidden="true"
            enable-background="new 0 0 48 48"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 48 48"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
              fill="#FFC107"
            />
            <path
              d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
              fill="#FF3D00"
            />
            <path
              d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
              fill="#4CAF50"
            />
            <path
              d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
              fill="#1976D2"
            />
          </svg>
        </button>
        <button
          aria-label="Sign up with Twitter"
          class="chakra-button css-14axtgj"
          type="button"
        >
          <svg
            aria-hidden="true"
            color="#55ACEE"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(85, 172, 238);"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;
